### PR TITLE
Server

### PR DESCRIPTION
**Pull request by d4nieldev**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,28 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (inbarbenchaim) [inline pom.xml:1]: make sure with Amit pls its working in here project after the PR- she wrote it.

File: src/main/java/Domain/UserController.java [hunk 2]
```diff
@@ -1,4 +1,5 @@
 package Domain;
+
```
Comment (inbarbenchaim) [inline src/main/java/Domain/UserController.java:2]: _users
_passwordEncoder

File: src/main/java/Domain/UserController.java [hunk 3]
```diff
@@ -34,7 +35,7 @@ public boolean isCredentialsCorrect(User user, String password) {
         return this.passwordEncoder.matches(password, user.getEncodedPassword());
     }
 
-    public void register(String user_name, String password, String email) throws Exception{
+    public void register(String user_name, String password, String email) throws Exception {
```
Comment (inbarbenchaim) [inline src/main/java/Domain/UserController.java:38]: documents and logs pls

File: src/main/java/Server/Server.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@SpringBootApplication
+@RestController
+@ComponentScan({ "Server", "ServiceLayer" })
+public class Server {
```
Comment (inbarbenchaim) [inline src/main/java/Server/Server.java:12]: // The purpose of the class: ......
why its important to the flow??

File: src/main/java/ServiceLayer/Response.java [hunk 5]
```diff
@@ -1,39 +1,33 @@
 package ServiceLayer;
+
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/Response.java:2]: pls add the main reason to use this class

File: src/main/java/ServiceLayer/UserService.java [hunk 6]
```diff
@@ -1,57 +1,62 @@
...
         tokenService = new TokenService();
     }
 
-    public Response logIn(String userName, String password) throws Exception {
+    public Response logIn(String userName, String password) {
         Response response = new Response();
         try {
             userController.logIn(userName, password);
             logger.info("User logged in: " + userName);
-            response.setReturnValue(tokenService.generateToken(userName));
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/UserService.java:22]: keep that name pls. its better for functions.

File: src/main/java/ServiceLayer/UserService.java [hunk 7]
```diff
@@ -1,57 +1,62 @@
...
+            response.set_returnValue(tokenService.generateToken(userName));
         } catch (Exception e) {
-            response.setErrorMessage("LogIn failed: " + e.getMessage());
+            response.set_errorMessage("LogIn failed: " + e.getMessage());
```
Comment (inbarbenchaim) [inline src/main/java/ServiceLayer/UserService.java:28]: also the name here

File: src/main/java/Server/UserMappings.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import ServiceLayer.*;
+
+@RestController
+@RequestMapping(path = "/api/user")
```
Comment (inbarbenchaim) [inline src/main/java/Server/UserMappings.java:12]: what this class is using for?

File: pom.xml (hunk 1 above)
Comment (d4nieldev) [reply pom.xml:1]: It just fixes a warning, specifies the encoding for different systems. It should work, but we should try it on other platforms.

File: src/main/java/Domain/UserController.java (hunk 2 above)
Comment (d4nieldev) [reply src/main/java/Domain/UserController.java:2]: I didn't change this file, just added the package.

File: src/main/java/Domain/UserController.java (hunk 3 above)
Comment (d4nieldev) [reply src/main/java/Domain/UserController.java:38]: This class is not my responsibility, read my commit messages, I have an auto formatter that just fixed spacing

File: src/main/java/Server/Server.java (hunk 4 above)
Comment (d4nieldev) [reply src/main/java/Server/Server.java:12]: I think it is clear from the name... Too much documentation is bad

File: src/main/java/ServiceLayer/Response.java (hunk 5 above)
Comment (d4nieldev) [reply src/main/java/ServiceLayer/Response.java:2]: Again, I think this is clear from the name

File: src/main/java/ServiceLayer/UserService.java (hunk 6 above)
Comment (d4nieldev) [reply src/main/java/ServiceLayer/UserService.java:22]: What name?

File: src/main/java/ServiceLayer/UserService.java (hunk 7 above)
Comment (d4nieldev) [reply src/main/java/ServiceLayer/UserService.java:28]: Ok got it

**Review by inbarbenchaim — APPROVED** [review]

**Pull request merged**